modin/core/storage_formats/pandas/parsers.py

```diff
@@ -653,9 +674,49 @@ def parse(fname, **kwargs):
         columns = [c for c in columns if c not in df.index.names and c in df.columns]
         if columns is not None:
             df = df[columns]
-        # Append the length of the index here to build it externally
         return _split_result_for_readers(0, num_splits, df) + [idx, df.dtypes]
 
+    @staticmethod
+    @doc(
+        _doc_parse_func,
+        parameters="""files_for_parser : list
+    List of files to be read.
+storage_options : dict
+    Parameters for specific storage engine.""",
+    )
+    def parse_fsspec_files(files_for_parser, storage_options, **kwargs):
+        from pyarrow.parquet import ParquetFile
+
+        columns = kwargs.get("columns", None)
+
+        chunks = []
+        for file_for_parser in files_for_parser:
+            if isinstance(file_for_parser.path, IOBase):
+                context = nullcontext(file_for_parser.path)
+            else:
+                context = fsspec.open(file_for_parser.path, **storage_options)
+            with context as f:
+                chunk = (
+                    ParquetFile(f)
+                    .read_row_groups(
+                        list(
```

Comment:
why do you want to materialize `range()` here? in most cases it just eats memory and slows down code

Reply:
We are trying to parallelize by reading over row groups so this function is only called when the partition is being materialized. We need to do this to get the resulting dataframe.

Reply:
@vnlitvinov I don't see any other way to [read](https://arrow.apache.org/docs/python/generated/pyarrow.parquet.ParquetFile.html#pyarrow.parquet.ParquetFile.read_row_groups) a range of row groups from a parquet file. It seems very unlikely that the number of groups in a single partition will be over 1000, so I think this will be fast enough.

Reply:
you don't have to call `list(range(x))` to iterate over `range(x)`... my question is why you want to call `list()`, why `range()` is not enough?

Reply:
`read_row_groups` takes a list, not an iterable it seems